test/hashing.spec.js

```diff
@@ -0,0 +1,1347 @@
+/**
```

Comment:
Please split this large file, either by source file tested, or by top-level `describe` to make it easier to understand for new users.